tasklist/client/src/v1/api/useAssignTask.mutation.tsx

```diff
@@ -71,6 +99,29 @@ function useAssignTask() {
 
       error.name = errorResult.data.message.title;
 
+      if (isTaskTimeoutError(errorResult.data.message)) {
+        const currentTask = client.getQueryData(
+          getUseTaskQueryKey(taskId),
+        ) as Task;
+        if (currentTask) {
+          client.setQueryData(getUseTaskQueryKey(taskId), {
+            ...currentTask,
+            taskState: 'ASSIGNING' as const,
+            assignee: null,
+          });
+        }
```

Comment:
No casting. This casting is already hiding an error since `currentTask` might be undefined too

This way it's properly typed

```suggestion
      const currentTask = client.getQueryData<Task>(
          getUseTaskQueryKey(taskId),
        );
        if (currentTask) {
          client.setQueryData(getUseTaskQueryKey(taskId), {
            ...currentTask,
            taskState: 'ASSIGNING',
            assignee: null,
          });
        }
```